artifactory/services/utils/artifactoryutils.go

```diff
@@ -517,6 +535,71 @@ func createPrioritiesFiles() ([]*content.ContentWriter, error) {
 	return []*content.ContentWriter{firstPriority, secondPriority, thirdPriority}, nil
 }
 
+func GetBuildInfo(buildName, buildNumber, projectKey string, flags CommonConf) (pbi *buildinfo.PublishedBuildInfo, found bool, err error) {
+	// Resolve LATEST build number from Artifactory if required.
+	name, number, err := GetBuildNameAndNumberFromArtifactory(buildName, buildNumber, flags)
+	if err != nil {
+		return nil, false, err
+	}
+
+	// Get build-info json from Artifactory.
+	httpClientsDetails := flags.GetArtifactoryDetails().CreateHttpClientDetails()
+	restApi := path.Join("api/build/", name, number) + GetProjectQueryParam(projectKey)
+	requestFullUrl, err := BuildArtifactoryUrl(flags.GetArtifactoryDetails().GetUrl(), restApi, make(map[string]string))
+
+	httpClient, err := flags.GetJfrogHttpClient()
+	if err != nil {
+		return nil, false, err
+	}
+	log.Debug("Getting build-info from: ", requestFullUrl)
+	resp, body, _, err := httpClient.SendGet(requestFullUrl, true, &httpClientsDetails)
+	if err != nil {
+		return nil, false, err
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		log.Debug("Artifactory response: " + resp.Status + "\n" + utils.IndentJson(body))
+		return nil, false, nil
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, false, errorutils.CheckError(errors.New("Artifactory response: " + resp.Status + "\n" + utils.IndentJson(body)))
+	}
+
+	// Build BuildInfo struct from json.
+	publishedBuildInfo := &buildinfo.PublishedBuildInfo{}
+	if err := json.Unmarshal(body, publishedBuildInfo); err != nil {
+		return nil, true, err
+	}
+
+	return publishedBuildInfo, true, nil
+}
+
+// Recursively, aggregate all transitive builds of the input buildName and buildNumber.
+// Build B is considered transitive of build A if the 2 following conditions are met:
+// 1. B is a submodule of A or another build that is a transitive of A (direct or indirect descendant).
+// 2. B is a module with "Build" type.
+func getAggregatedBuilds(buildName, buildNumber, projectKey string, flags CommonConf) ([]Build, error) {
+	buildInfo, _, err := GetBuildInfo(buildName, buildNumber, projectKey, flags)
+	if err != nil || buildInfo == nil {
+		return []Build{}, err
+	}
+	aggregatedBuilds := []Build{{
+		BuildName:   buildName,
+		BuildNumber: buildNumber,
+	}}
+	for _, module := range buildInfo.BuildInfo.Modules {
+		if module.Type == buildinfo.Build {
+			buildSplit := strings.Split(module.Id, "/")
```

Comment:
I think that istead of the above ```Split``` function, we should use the ```parseNameAndVersion``` function, which takes care of a few edge cases.